vector/src/main/java/im/vector/app/core/extensions/BottomSheetDialog.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.extensions
+
+import android.os.Build
+import android.util.DisplayMetrics
+import androidx.annotation.FloatRange
+import androidx.annotation.RequiresApi
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+fun BottomSheetDialog.setPeekHeightAsScreenPercentage(@FloatRange(from = 0.0, to = 1.0) percentage: Float) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        setPeekHeightPostApi30(percentage)
+    } else {
+        setPeekHeightPreApi30(percentage)
+    }
+}
```

Comment:
I would rather create an extension `getScreenheightPx()` to get the Window height, which will be more reusable.
Then just call 
```
val height = window?.getScreenheightPx()
behavior.setPeekHeight((height * percentage).toInt(), true)
```
 here.

We have some equivalent code at this point: https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/roomprofile/uploads/media/RoomUploadsMediaFragment.kt#L86

It would deserve a quick refactor to extract useful and reusable and compatible extensions, in the package `im.vector.app.core.extensions`. Happy to do it later if you do not have time.